### PR TITLE
Update syn and quote

### PR DESCRIPTION
**Pull request by dimbleby**

### CODE REVIEW

File: slog-extlog-derive/src/lib.rs [hunk 1]
```diff
@@ -708,36 +738,47 @@ fn parse_stat_trigger(attr_val: &[syn::NestedMetaItem], body: &syn::Body) -> Sta
...
             _ => panic!("Invalid format for StatTrigger - parameters must be key-value pairs"),
         };
 
-        match name {
-            "StatName" => {
-                id = Some(syn::parse_ident(val).expect("Could not parse name in StatTrigger"))
-            }
-            "Condition" => {
-                cond = Some(syn::parse_expr(val).expect("Could not parse condition in StatTrigger"))
+        match &name.as_ref() {
```
Comment (mthebridge) [inline slog-extlog-derive/src/lib.rs]: Can you not drop all the `&`s here?

File: slog-extlog-derive/src/lib.rs [hunk 2]
```diff
@@ -697,7 +724,10 @@ fn is_attr_stat_id(attr: &syn::Attribute, id: &syn::Ident) -> bool {
 }
 
 // Parses the StatTrigger attribute.
-fn parse_stat_trigger(attr_val: &[syn::NestedMetaItem], body: &syn::Body) -> StatTriggerData {
+fn parse_stat_trigger<'a, T>(attr_val: T, body: &syn::Data) -> StatTriggerData
```
Comment (mthebridge) [inline slog-extlog-derive/src/lib.rs]: I'd have a mild preference for using `impl Iterator<...>` here as slightly more concise (and we don't really care what T is beyond that) over adding a generic parameter, but maybe that doesn't work with the lifetime?

**Review by mthebridge — COMMENTED** [review]
LGTM - tahnks.  Couple of cosmetic things.

File: slog-extlog-derive/src/lib.rs (hunk 1 above)
Comment (dimbleby) [reply slog-extlog-derive/src/lib.rs]: yes, I wonder why I put them in!

File: slog-extlog-derive/src/lib.rs (hunk 2 above)
Comment (dimbleby) [reply slog-extlog-derive/src/lib.rs]: that works, I'll change

**Pull request merged**